docs/shadow_paths.md

```diff
@@ -0,0 +1,101 @@
+    OperStatus       E_Interface_OperStatus             `path:"state/oper-status" module:"openconfig-interfaces/openconfig-interfaces"`
+}
+```
+
+The aim of this document is to clarify what shadow paths are and why they exist
+as a generation option.
+
+## Shadow paths: compressed-out "config" or "state" YANG leaves
+
+https://datatracker.ietf.org/doc/html/draft-openconfig-netmod-opstate-01#section-2
+contains a diagram of the relationship between intended config (`/config`) and
+applied config (`/state`) leaves:
+
+```
+          +---------+
+          |         |    transition intended
+          |intended |    to applied
+          | config  +---------+
+          |         |         |
+          +---------+         |
+              ^               |         config: true
+   +----------|------------------------------------+
+              |               |         config: false
+              |               |
+              |               |
+              |       +-----------------------------+
+              |       |       | operational state   |
+              |       |  +----v----+ +-----------+  |
+              |       |  |         | |           |  |
+              +       |  | applied | |  derived  |  |   operational:true
+            same +------>| config  | |   state   |<-------+
+            leaves    |  |         | |           |  |
+                      |  |         | |           |  |
+                      |  +---------+ +-----------+  |
+                      +-----------------------------+
+```
+
+In this diagram, applied config are "config false", or `/state` leaves that
+mirror intended config leaves, which are "config true", or `/config` leaves. Per
+[design.md](design.md#openconfig-path-compression), one of these are compressed
+out depending on the value of the `-prefer_operational_state` generation flag.
+
+Shadow paths are relevant only for compressed GoStructs, and indicate the
+compressed-out `/config` or `/state` YANG `leaf` nodes.
+
+## Problems with Path Compression and how Shadow Paths Help
+
+Path compression leads to the GoStruct not being able to represent both intended
+config and applied config at the same time. This leads to two problems:
+
+1.  When subscribing to a non-leaf path, some gNMI clients want to silently
+    ignore the compressed-out paths rather than erroring out due to an
+    unrecognized path.
+2.  Some use cases (e.g. [ygnmi](https://github.com/openconfig/ygnmi#queries))
+    use the same compressed GoStruct for representing either the "config view"
+    (intended config+derived state) combination, or the "state view" (applied
+    config+derived state) combination of leaves. We want to enable such a
+    meaning-switch for certain ygot utilities (e.g. marshalling/unmarshalling).
```

Comment:
By meaning switch here, do you mean allow switching between `config` views and `state` views?

Reply:
Yes, changed wording verbatim.